app/controllers/api/v2/smart_proxies_controller.rb

```diff
@@ -65,6 +65,17 @@ def refresh
         process_response @smart_proxy.refresh.blank? && @smart_proxy.save && @smart_proxy.reload
       end
 
+      api :POST, "/smart_proxies/:id/import_subnets", N_("Import subnets from Smart proxy")
+      param :id, String, :required => true
+
+      def import_subnets
+        proxy = SmartProxy.find(params[:id])
+        @subnets = Subnet::Ipv4.import(proxy)
+        if @subnets.empty?
+          render_message(_("No new subnets found"), :status => :unprocessable_entity)
```

Comment:
Is this really an `unprocessable_entity`? I'd say it's not a correct response. HTTP 422 says the sender did something wrong (all 4xx errors really), but in this case I'd say it's actually just a normal operation.

Reply:
I think  I can't return 200 because on the hammer side the returned message will be: " Import subnets succeeded". 
I agree that unprocessable_entity is not accurate, and I think 404 is a better option.

Reply:
I'd encourage you to read the HTTP status codes. Especially https://developer.mozilla.org/en-US/docs/Web/HTTP/Status gives a good overview. 4xx implies a client error, but IMHO they did nothing wrong if there is no change.

What I'd suggest instead is to treat it as a success. If you need to distinguish the two cases I'd suggest:
* 200 for no change
* 201 for a new subnet (or multiple) were imported.

Quoting the earlier link for HTTP 201:
> The request succeeded, and a new resource was created as a result. This is typically the response sent after POST requests, or some PUT requests.

This is exactly what happens.

For no change you can also consider HTTP 204 No Content.

Reply:
@adiabramovitch, don't be worry about what hammer will show on 200 status code, we can always adjust/adapt stuff in hammer. The more important thing is to consider that API can be used directly and we must follow the specifications of the HTTP status codes. And regarding this I agree with Ewoud: we can use 200/204 and 201 (this way we could actually verify how hammer deals with this and address any implications accordingly)